pkg/oc/cli/cmd/set/deployment_strategy.go

```diff
@@ -0,0 +1,483 @@
+	infos := o.Infos
+	singleItemImplied := len(o.Infos) <= 1
+	if o.Builder != nil {
+		loaded, err := o.Builder.Do().IntoSingleItemImplied(&singleItemImplied).Infos()
+		if err != nil {
+			return err
+		}
+		infos = loaded
+	}
+
+	patches := CalculatePatches(infos, o.Encoder, func(info *resource.Info) (bool, error) {
+		switch t := info.Object.(type) {
+		case *deployapi.DeploymentConfig:
+			return o.updateDeploymentConfig(t.DeepCopy())
+		case *extensions.Deployment:
+			return o.updateDeployment(t.DeepCopy())
+		case *extensions.DaemonSet:
+			return o.updateDaemonSet(t.DeepCopy())
+		case *apps.StatefulSet:
+			return o.updateStatefulSet(t.DeepCopy())
+		default:
+			return false, nil
+		}
+	})
+
+	if singleItemImplied && len(patches) == 0 {
+		return fmt.Errorf("%s/%s does not have an applicable strategy", infos[0].Mapping.Resource, infos[0].Name)
+	}
+
+	if len(o.Output) > 0 || o.Local || kcmdutil.GetDryRunFlag(o.Cmd) {
+		return o.PrintObject(infos)
+	}
+
+	failed := false
+	for _, patch := range patches {
+		info := patch.Info
+		if patch.Err != nil {
+			fmt.Fprintf(o.Err, "error: %s/%s %v\n", info.Mapping.Resource, info.Name, patch.Err)
+			continue
+		}
+
+		if string(patch.Patch) == "{}" || len(patch.Patch) == 0 {
+			fmt.Fprintf(o.Err, "info: %s %q was not changed\n", info.Mapping.Resource, info.Name)
+			continue
+		}
+
+		obj, err := resource.NewHelper(info.Client, info.Mapping).Patch(info.Namespace, info.Name, types.StrategicMergePatchType, patch.Patch)
+		if err != nil {
+			fmt.Fprintf(o.Err, "error: %v\n", err)
+			failed = true
+			continue
+		}
+
+		info.Refresh(obj, true)
+		kcmdutil.PrintSuccess(o.Mapper, o.ShortOutput, o.Out, info.Mapping.Resource, info.Name, false, "updated")
+	}
+	if failed {
+		return kcmdutil.ErrExit
+	}
+	return nil
+}
+
+func (o *DeploymentStrategyOptions) applyDeploymentConfigRecreateStrategyOptions(s *deployapi.RecreateDeploymentStrategyParams) bool {
+	if o.TimeoutSeconds != deployapi.DefaultRecreateTimeoutSeconds {
+		s.TimeoutSeconds = &o.TimeoutSeconds
+		return true
+	}
+	return false
+}
+
+func (o *DeploymentStrategyOptions) applyDeploymentConfigRollingStrategyOptions(s *deployapi.RollingDeploymentStrategyParams) bool {
+	updated := false
+	if o.TimeoutSeconds != deployapi.DefaultRecreateTimeoutSeconds {
+		s.TimeoutSeconds = &o.TimeoutSeconds
+		updated = true
+	}
+	if len(o.MaxSurge) > 0 {
+		s.MaxSurge = intstr.FromString(o.MaxSurge)
+		updated = true
+	}
+	if len(o.MaxUnavailable) > 0 {
+		s.MaxUnavailable = intstr.FromString(o.MaxUnavailable)
+		updated = true
+	}
+	// TODO: This has potential to reset the pre-existing values for these two
+	// fields if they are already set to custom values. We need to pick if this
+	// command will override those and set them back to default values if used or
+	// not.
+	if o.IntervalSeconds != deployapi.DefaultRollingIntervalSeconds || (s.IntervalSeconds != nil && *s.IntervalSeconds != deployapi.DefaultRollingIntervalSeconds) {
```

Comment:
I don't think we should be checking against default values here and they can also differ between API groups they were created in

Reply:
yeah, I dunno what should be the best way to do this... not sure if flags can have "not-set" status?

Reply:
make them pointers? the methods end with `P`

Reply:
@tnozicka no, P just adds the shortcut...

// Int64VarP is like Int64Var, but accepts a shorthand letter that can be used after a single dash.

Reply:
.Changed()